### PR TITLE
Add readthedocs config file

### PR DESCRIPTION
**Pull request by jasonpaulos**
Since merging #269, our readthedocs build has failed (as an example, see https://readthedocs.org/projects/pyteal/builds/16635690/), and I believe adding this configuration will properly tell it to use Python 3.10 to build our docs.

If this works, a similar thing will likely need to be added to the Python SDK following https://github.com/algorand/py-algorand-sdk/pull/313 (cc @algochoi)

### CODE REVIEW

**Review by michaeldiamant — APPROVED** [review]
@jasonpaulos Thanks for catching!

I'll follow up out-of-band to understand how to better monitor the build.

A few thoughts:
* Reading the docs more closely, I see it specifies `.yaml` extension.  We previously defined `.readthedocs.yml`.  Presumably the file extension explains the build failure.
* I don't feel strongly about making the change in this PR, but it feels like we ought to delete `.readthedocs.yml`.
 * Additionally, `.readthedocs.yml` explicitly specifies `python.version: 3.10`.  From my read of the docs, `build.os.tools` _seems equivalent.  Though it feels prudent to set both once the build works (or to make it work).

File: .readthedocs.yaml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+   configuration: docs/conf.py
+
+# If using Sphinx, optionally build your docs in additional formats such as PDF
+# formats:
+#    - pdf
+
+# Optionally declare the Python requirements required to build your docs
+python:
+   install:
+   - requirements: docs/requirements.txt
```
Comment (tzaffi) [inline .readthedocs.yaml:29]:
```suggestion
   - requirements: docs/requirements.txt

```

**Review by tzaffi — APPROVED** [review]
LGTM - (even without the suggested extra line in `.readthedocs.yaml`)

**Comment by algochoi** [comment]
Can this yaml file be copied over to the `py-algorand-sdk` directly? It looks like that repo doesn't define one yet.

**Comment by jasonpaulos** [comment]
> Can this yaml file be copied over to the `py-algorand-sdk` directly? It looks like that repo doesn't define one yet.

Likely yes

**Pull request merged**